src/sources/soundsourcestem.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "sources/soundsourceffmpeg.h"
+#include "sources/soundsourceprovider.h"
+#include "util/samplebuffer.h"
+
+namespace mixxx {
+
+/// @brief Handle a single stem stem embedded in a stem file
+class SoundSourceSingleSTEM : public SoundSourceFFmpeg {
+  public:
+    // streamIdx is the FFmpeg stream id, which may different than stemIdx + 1
+    // because STEM may contain other non audio stream
+    explicit SoundSourceSingleSTEM(const QUrl& url, unsigned int streamIdx);
+
+  protected:
+    OpenResult tryOpen(
+            OpenMode mode,
+            const OpenParams& params) override;
+
+  private:
+    unsigned int m_streamIdx;
+};
+
+/// @brief Handle a stem file, composed of multiple audio channel. Can open in
+/// stereo or in stem (4 x stereo)
```

Comment:
Can the the user of the sound source decide? Probably not. Can you refine the comment?

We have the issue that pure mp4 files are opened with libfaad which is not offset compatible. This means we will have the issue that before merging this PR *.stem.mp4 have been decoded with a different offset and the analysis is no longer consistent. We need to deal with it somehow. Suggestions?

Reply:
Since we can (in future) load a STEM file as well into a deck (all 4 stems seperated) as into a sampler/preview deck (either mixed stereo OR a selected single stem stereo track), STEM files need always to be analyzed with the same offset, independend if we use the mixed stereo, or the stem tracks.
While non-STEM .mp4 are completly independent.

I think we are safe, if we can safely distinguish STEM files from non-STEM .mp4 files and read STEM files always using FFMPEG.

Reply:
@JoergAtGithub is correct - when you open a SoundSource, you pass the `OpenParams` which allow to request a maximum number of supported channel (since Mixxx uses the StereoProxy in case the channel count is less than requested). This allows decks which must not use STEM for performance or usability reason to use the soundsource as a strict stereo (samples, preview, but also analyser currently, till the PR which introduce STEM analysis)